api_trial_system.yaml

```diff
@@ -0,0 +1,151 @@
+swagger: '2.0'
+info:
+  version: 13.35.0
+  title: Trial System API
+host: localhost
+basePath: /trials/api/v1
+schemes:
+  - https
+security:
+  - Bearer: []
+securityDefinitions:
+  Bearer:
+    type: apiKey
+    name: Authorization
+    in: header
+paths:
+  /trials/{trialId}/subscriptions:
+    parameters:
+      - $ref: '#/parameters/pathTrialId'
+    get:
+      operationId: getSubscription
+      summary: Get subscription detail
+      description: >-
+        Returns the details about the subscription of the authenticated user for
+        the specified trial.
+      responses:
+        '200':
+          description: Success.
+          schema:
+            $ref: '#/definitions/Subscription'
+        '401':
+          description: Authorization error.
+        '404':
+          description: Resource not found.
+        '429':
+          description: Too may requests.
+        '500':
+          description: There was an error.
+          schema:
+            $ref: '#/definitions/ProblemJson'
+    post:
+      operationId: createSubscription
+      summary: Create a subscription
+      description: Subscribe the authenticated user to the given trial.
+      responses:
+        '201':
+          description: Success.
+          schema:
+            $ref: '#/definitions/Subscription'
```

Comment:
**question**: should we add the `202` status code?

This will help the caller to distinguish between "the resources has been created" and "ok, i'm handling the request".
What do you think?

In this case, if we are going to add the `202` status code, then I'd evaluate to convert the `createdAt` parameter to an optional parameter (this is valid if we are going to return the same schema `'#/definitions/Subscription'`)

Reply:
Yes that true 👍

Reply:
Done ✅